extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -47,74 +49,91 @@
   private static final Log logger = LogFactory.getLog( ActionInvokerSystemListener.class );
   private static final String WORK_ITEM_FILE_EXTENSION = ".json";
   private static final String DEFAULT_CONTENT_FOLDER = "system/default-content";
+  private static final String ERROR = "error";
+  private static final String OK = "ok";
+  private static final String WI_STATUS = "wi-status.";
   private String environmentVariablesFolder;
-
   //for unit testability
   String getSolutionPath( ) {
     return PentahoSystem.getApplicationContext().getSolutionPath( DEFAULT_CONTENT_FOLDER );
   }
 
+  //places a failure breadcrumb in the specified folder
+  private void placeBreadcrumbFile( String folderPath, boolean success ) {
+    String ext = ERROR;
```

Comment:
Just a suggestion for cleaner / shorthand code:
String ext = success ? OK : ERROR;